### PR TITLE
Integrate Mercattrick statistics on transfers list page (skills table and player box)

### PR DESCRIPTION
**Pull request by dromichet**
Resolves minj/foxtrick#1656

### CODE REVIEW

**Comment by minj** [comment]
Sorry for the delay. I hope to have time for FT next week

**Comment by dromichet** [comment]
No problem! I was just curious if the implementation is ok or do I need to change anything.

**Comment by minj** [comment]
I've looked through the code and can't see any major issues for now. Detail review will follow later.

I see that you've added CORS headers, that's a nice workaround until I get the hold on myself and reimplement the permission system :)

File: content/api/mercattrick/common.js [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+ * Low-level function to generate requests to Mercattrick and process the response.
+ *
+ * Leaves Mercattrick alone if trouble is detected. Should be used with all apis.
+ *
+ * @param  {string}          api    api name for logging purposes
+ * @param  {string}          url    api URL
+ * @return {Promise<string|FetchError>} rejects with { status, text }
+ */
+Foxtrick.api.mercattrick._fetchOrIgnore = async (api, url) => {
+	const MSEC = Foxtrick.util.time.MSEC;
```
Comment (minj) [inline content/api/mercattrick/common.js:124]: Should be `Foxtrick.util.time.MSECS_IN_SEC`

File: content/foxtrick.properties [hunk 2]
```diff
@@ -503,6 +503,20 @@ module.U20LastMatch.SeniorPlayers.desc=Show last official U20 match in the Senio
...
+module.MercattrickStats.desc=Display how many managers are interested in a transfer listed player.
+module.MercattrickStats.enableTLPage.desc=Enable on transfer search result page.
+
+MercattrickStats.filters=Mercattrick Filters
+MercattrickStats.filters.abbr=mtF
+MercattrickStats.bookmarks=Mercattrick Bookmarks
+MercattrickStats.bookmarks.abbr=mtB
+MercattrickStats.transfers=Mercattrick Stats:
+MercattrickStats.filtersAndBookmarks=Filters: %1 / Bookmarks: %2
+MercattrickStats.api.down=hattrick youthclub service is down today
```
Comment (minj) [inline content/foxtrick.properties:516]: needs to be corrected

File: content/foxtrick.properties [hunk 3]
```diff
@@ -503,6 +503,20 @@ module.U20LastMatch.SeniorPlayers.desc=Show last official U20 match in the Senio
 module.U20LastMatch.AllPlayers.desc=Show last official U20 match in the Senior/Youth/NT Players page.
 module.U20LastMatch.TransfersResults.desc=Show last official U20 match in the Transfers Search Results page.
 
+# Mercattrick
+module.MercattrickStats.desc=Display how many managers are interested in a transfer listed player.
+module.MercattrickStats.enableTLPage.desc=Enable on transfer search result page.
```
Comment (minj) [inline content/foxtrick.properties:508]: Any plans to use this on other pages?

File: content/information-aggregation/mercattrick-stats.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+                const bookmarks = statObj ? statObj.bookmarks_count : 0;
+                // creating element
+                const container = Foxtrick.createFeaturedElement(doc, module, 'div');
+
+                Foxtrick.addClass(container, 'ft-mercattrick-stats');
+
+                let img = doc.createElement('img');
+                img.width = 21;
+                img.height = 16;
+                img.title = Foxtrick.L10n.getString('MercattrickStats.title');
```
Comment (minj) [inline content/information-aggregation/mercattrick-stats.js:59]: `aria-label` missing

`Foxtrick.addImage` is to be used for backwards- and forwards-compatibility with async resource contexts

File: content/resources/css/mercattrick-stats.css [hunk 5]
```diff
@@ -0,0 +1,8 @@
+.ft-mercattrick-stats {
+	margin: 10px 6px;
+}
+
+.ft-mercattrick-stats img {
+	vertical-align: middle;
+	padding-right: 5px;
```
Comment (minj) [inline content/resources/css/mercattrick-stats.css:7]: use either `padding-inline-end` or an rtl block

**Review by minj — CHANGES_REQUESTED** [review]
Overall great code. Minor fixes needed.

Code style is somewhat inconsistent.

Let me know, if you will be addressing my comments yourself.

Alternatively, GitHub now has a feature to allow access for the repo owner (i.e. me) to push into a PR branch on the fork (i.e. your repo)

**Comment by minj** [comment]
Apologies for the delay. I was not in a good place mentally for weeks.

File: content/foxtrick.properties (hunk 3 above)
Comment (dromichet) [reply content/foxtrick.properties:508]: I think I've added it to all the appropriate pages, so the answer is no, for now. :)

**Comment by dromichet** [comment]
Hi @minj.
I'm sad to hear that, but I hope everything is fine now!

Regarding your review, since I'm not very sure about all the changes that I need to do, I've sent you an access invitation to my repo. I don't know if I need to grant you any other rights, so let me know.

File: content/api/mercattrick/players-stats.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+if (!Foxtrick.api.mercattrick.URL)
+	Foxtrick.api.mercattrick.URL = {};
+
+Foxtrick.api.mercattrick.URL.playersStats = '/players/stats/';
+
+/**
+ * Tries to fetch the player stats from Mercattrick and returns a Promise for players
+ *
+ * @param  {Array}  playerIds players ID to fetch data for
+ * @return {Promise<Stats>}
```
[the remark this answers is not in the document]
Comment (dromichet) [reply content/api/mercattrick/players-stats.js:46]: Indeed, it should have been StatsList.

**Review by minj — APPROVED** [review]

**Comment by minj** [comment]
Thanks for your contribution

**Comment by dromichet** [comment]
I was just about to finish the remaining issues when I noticed you already merged it.
Thanks @minj.

**Comment by minj** [comment]
@dromichet you've forgot `MercattrickStats.title` l10n string ;)

**Pull request merged**